common/sketch/src/main/java/org/apache/spark/util/sketch/CountMinSketchImpl.java

```diff
@@ -368,4 +379,30 @@ public static CountMinSketchImpl readFrom(InputStream in) throws IOException {
 
     return new CountMinSketchImpl(depth, width, totalCount, hashA, table);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+      this.writeTo(bos);
+      byte[] bytes = bos.toByteArray();
+      out.writeObject(bytes);
+    }
+  }
+
+  @Override
+  public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+    byte[] bytes = (byte[]) in.readObject();
+
+    try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes)) {
+      CountMinSketchImpl sketch = CountMinSketchImpl.readFrom(bis);
+
+      this.depth = sketch.depth;
```

Comment:
it'd be good to refactor this so we don't need to assign the variables. one way is to take the serialization/deserialization code out of readFrom into a function.